sky/packages/sky/lib/rendering/proxy_box.dart

```diff
@@ -295,18 +295,12 @@ class RenderOpacity extends RenderProxyBox {
   void paint(PaintingContext context, Offset offset) {
     if (child != null) {
       int a = _alpha;
-
       if (a == 0)
         return;
-
-      if (a == 255) {
+      if (a == 255)
         context.paintChild(child, offset.toPoint());
-        return;
-      }
-
-      context.canvas.saveLayer(null, _paint); // TODO(abarth): layerize
-      context.paintChild(child, offset.toPoint());
-      context.canvas.restore();
+      else
+        context.paintChildWithPaint(child, offset.toPoint(), null, _paint);
```

Comment:
In a future patch, we should split up paintChildWithPaint into paintChildWithOpacity and paintChildWithColorFilter.